### DISCUSSION TITLE
Running version 1.4.1 on Vista

### DISCUSSION THREAD

**Discussion by LordLynix** [diasurgical/DevilutionX#6062] [Q&A] [answered]
Hello, I'd really like to be able to play this fantastic port on my oldies gaming PC with x64 Vista. I know - it's an old system that nobody likes. Regardless, it would be very helpful if someone could please provide me with some assistance. Version 1.3.0 works, but it'd be great to enjoy the bugfixes added in the later builds. As far as I know, the problem is with certain new features introduced after version 1.3.0 which set off a KERNEL32.dll!SetThreadErrorMode error. Is it possible to build the 1.4.1 version without those?

Any assisstance would be much appreciated!

**Comment by AJenbo** [comment]
The issue is either Discord integration or ZeroTier ... or both. Without them 1.4.1 should work with XP+

**Reply by AJenbo** [reply]
@Trihedraf didn't you end up making a build like this for the people from Vogons?

**Reply by Trihedraf** [reply, marked as answer]
So for Vista it appears to be Discord integration that stops it from running, but ZeroTier works unlike XP where it is both.
[devilutionx_1.4.1_x64_Vista.zip](https://github.com/diasurgical/devilutionX/files/11360542/devilutionx_x64_Vista.zip)

**Reply by LordLynix** [reply]
Works like a charm! Thank you very much for your swift reply and for sharing the file!

**Comment by Gerwin2k** [comment]
On my website, in the back-ports section, is a Windows XP compatible v1.5.0 build. The v1.5.0 release of yesterday that is.
http://gb-homepage.nl/
(Disabled Zerotier and Discord integration. As well as Sodium packet encryption. May try to put sodium in, if requested. )

**Reply by AJenbo** [reply]
Thanks for updating that. Sodium is only required for private games but as validation has gotten better and TCP games aren't publicly announced I think it's ok to not have it in there. It will already be a lot safer then running the original.

**Reply by Trihedraf** [reply]
Sodium should work just fine with XP

**Reply by Gerwin2k** [reply]
Suppose it should. Just that the cmake package finder only passes with libsodium.dll.a, whilst I want it to use libsodium.a (static).

Disabling "Findsodium.cmake", I did just build a new DevilutionX executable with the static sodium lib. Don't see any difference, except the file size grew a little. Suppose it is fine though. Download package has been updated.

**Reply by Trihedraf** [reply]
All you have to do is add “-DDEVILUTIONX_SYSTEM_LIBSODIUM=OFF -DDEVILUTIONX_STATIC_LIBSODIUM=ON”
to your build command and it will from source build a static Libsodium

**Reply by Gerwin2k** [reply]
Thanks Trihedraf. These commands work well. I won't have to use my previous workaround anymore.

**Reply by AJenbo** [reply]
@Gerwin2k 1.5.1 is out now so if you have time to build it for XP I know a few people will appreciate it :)

**Reply by Gerwin2k** [reply]
Ah, I had not even received your message yet, but was already on it ;)

**Comment by LordLynix** [comment]
> On my website, in the back-ports section, is a Windows XP compatible v1.5.0 build. The v1.5.0 release of yesterday that is. http://gb-homepage.nl/ (Disabled Zerotier and Discord integration. As well as Sodium packet encryption. May try to put sodium in, if requested. )

Thank you for making this available!

**Comment by Gerwin2k** [comment]
Congratulations on the release of DevilutionX v1.5.1 
I just build and uploaded a Windows XP compatible binary of this version. It was the same procedure as before. Available here:
http://gb-homepage.nl/
